### ISSUE TITLE
invalid value for proxy (The proxy URL is not a valid socks url.)

### ISSUE THREAD

**Issue by joesonw** [pulumi/pulumi#11086]
## Steps to reproduce

1. connect to any mysql without setting `Proxy` input with go

Expected:  success

Actual:  invalid value for proxy (The proxy URL is not a valid socks url.)

**Comment by pierskarsenbarg** [comment]
Eng team suggested I transfer it to this repo

**Comment by pierskarsenbarg** [comment]
This is only an issue in Go. The following typescript works:

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as mysql from "@pulumi/mysql";

const stackref = new pulumi.StackReference("pierskarsenbarg/db-setup/dev");

const endpoint = stackref.getOutput("connectionString");
const password = stackref.getOutput("password");
const username = "piers";

const provider = new mysql.Provider("provider", {
    username,
    endpoint,
    password
})

const db = new mysql.Database("db", {}, {provider: provider});
```

**Comment by squaremo** [comment]
This came up again, well spotted @csakshaug. I logged the issue https://github.com/pulumi/pulumi-terraform-bridge/issues/635 to fix the underlying problem.

**Comment by justinvp** [comment]
This doesn't repro for me. Looks like it was fixed by #12976.
